File: c++ files/md2_t2/a.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node* left;
    struct node* right;
};
struct node* newnode(int data)
{
    struct node* temp = malloc(sizeof(struct node));
    temp->data = data;
    temp->left = NULL;
    temp->right = NULL;
    return temp;
}

struct node* insert(int i,int j,int a[])
{

    int mid = (i+j)/2;
    struct node* root = newnode(a[mid]);
    if(i==j)
        return root;
    if(i<mid){
        printf("left =");
        for(int k=i;k<mid;k++)
        printf("%d ",a[k]);
        printf("\n");
        root->left = insert(i,mid-1,a);
    }
    if(mid<j)
    {
        printf("right =");
        for(int k=mid+1;k<=j;k++)
            printf("%d ",a[k]);
        printf("\n");
        root->right = insert(mid+1,j,a);
    }
    return root;

}
void inorder(struct node* root)
{
    if(root==NULL)
        return;
    printf("%d ",root->data);
    inorder(root->left);
    inorder(root->right);
}
int main()
{
    int a[] = {1,2,3,4,5,6,7,8,9};int n = sizeof(a)/sizeof(a[0]);
    struct node* root = malloc(sizeof(struct node));
    root = insert(0,n-1,a);
    inorder(root);
}

File: c++ files/md2_t2/a.py
class Node:
    def __init__(self, value):
        self.value = value
        self.left = self.right = None
    

def insert(n):
    mid = len(n) // 2
    root = Node(n[mid])
    if len(n) > 1:
        root.left = insert(n[:mid])
        root.right = insert(n[mid+1:])
    return root

def print_tree(root):
    if root:
        print(root.value)
        print_tree(root.left)
        print_tree(root.right)

def insert2(i,j,n):
    mid = (i+j) // 2
    root = Node(n[mid])
    if i < j:
        print("left = ",n[i:mid])
        root.left = insert2(i,mid-1,n)
        print("right = ",n[mid+1:j+1])
        root.right = insert2(mid+1,j,n)
    return root

n = [1, 2, 3, 4, 5, 6, 7,8,9]
root = insert(n)
# root = insert2(0,len(n)-1,n)
print_tree(root)



